app/controllers/declaration/AdditionalProcedureCodesController.scala

```diff
@@ -76,10 +76,15 @@ class AdditionalProcedureCodesController @Inject()(
 
     def handleActionTypes(procedureCode: ProcedureCode, validAdditionalProcedureCodes: Seq[AdditionalProcedureCodeModel]) =
       formAction match {
-        case Add if !boundForm.hasErrors =>
-          addAnotherCodeHandler(mode, itemId, boundForm.get, procedureCode, cachedData, validAdditionalProcedureCodes)
-        case SaveAndContinue | SaveAndReturn if !boundForm.hasErrors =>
-          saveAndContinueHandler(mode, itemId, boundForm.get, procedureCode, cachedData, validAdditionalProcedureCodes)
+        case Add | SaveAndContinue | SaveAndReturn if !boundForm.hasErrors =>
+          validateForm(
+            formAction,
+            mode,
+            returnErrorPage(mode, itemId, boundForm.get, procedureCode, cachedData.additionalProcedureCodes, validAdditionalProcedureCodes),
```

Comment:
Nit: assign `returnErrorPage` to val for better readibility

Reply:
Good idea :+1: